sdk/tests/deqp/modules/shared/glsTextureTestUtil.js

```diff
@@ -923,7 +923,8 @@ glsTextureTestUtil.computeLodFromDerivates3D = function(mode, dudx, dvdx, dwdx,
             DE_ASSERT(false);
     }
 
-    return Math.log(p) * 1.44269504089;
+    // Native dEQP uses 32-bit numbers. So here 64-bit floating numbers should be transformed into 32-bit ones to ensure the correctness of the result.
+    return new Float32Array([Math.log(p)])[0] * new Float32Array([1.44269504089])[0];
```

Comment:
You know, I am investigating the precision related bugs too. A simple summary is that the precision of 'float' variable in C/C++ is not so accurate as that of var in JS. But the precision of var in JS is (almost the same with) not so accurate as that of 'double' variable in C/C++. 
A ranking of the accuracy: 
the accuracy of float variable in C/C++    < 
the accuracy of var in JS     <    the accuracy of double variable in C/C++.